src/pages/Settings/Settings.tsx

```diff
@@ -1,33 +1,47 @@
-import { useContext, useEffect, useState } from 'react'
+import { ChangeEvent, useContext, useEffect, useState } from 'react'
 import Box from '@mui/material/Box'
 import Button from '@mui/material/Button'
 import Typography from '@mui/material/Typography'
 import Divider from '@mui/material/Divider'
+import { Switch } from '@mui/material'
 
 import { ShellContext } from 'contexts/ShellContext'
 import { StorageContext } from 'contexts/StorageContext'
 import { PeerNameDisplay } from 'components/PeerNameDisplay'
 
 import { ConfirmDialog } from '../../components/ConfirmDialog'
+import { SettingsContext } from '../../contexts/SettingsContext'
 
 interface SettingsProps {
   userId: string
 }
 
 export const Settings = ({ userId }: SettingsProps) => {
   const { setTitle } = useContext(ShellContext)
+  const { updateUserSettings, getUserSettings } = useContext(SettingsContext)
   const { getPersistedStorage } = useContext(StorageContext)
   const [
     isDeleteSettingsConfirmDiaglogOpen,
     setIsDeleteSettingsConfirmDiaglogOpen,
   ] = useState(false)
+  const [playSoundOnNewMessage, setPlaySoundOnNewMessage] = useState(
+    () => getUserSettings().playSoundOnNewMessage
+  )
```

Comment:
I think we may be able to eliminate these state variables and just use `getUserSettings().playSoundOnNewMessage` directly. Perhaps something like this:

```diff
diff --git a/src/pages/Settings/Settings.tsx b/src/pages/Settings/Settings.tsx
index b9842ca..8c88821 100644
--- a/src/pages/Settings/Settings.tsx
+++ b/src/pages/Settings/Settings.tsx
@@ -24,9 +24,7 @@ export const Settings = ({ userId }: SettingsProps) => {
     isDeleteSettingsConfirmDiaglogOpen,
     setIsDeleteSettingsConfirmDiaglogOpen,
   ] = useState(false)
-  const [playSoundOnNewMessage, setPlaySoundOnNewMessage] = useState(
-    () => getUserSettings().playSoundOnNewMessage
-  )
+  const { playSoundOnNewMessage } = getUserSettings()
 
   const persistedStorage = getPersistedStorage()
 
@@ -38,7 +36,6 @@ export const Settings = ({ userId }: SettingsProps) => {
     event: ChangeEvent,
     value: boolean
   ) => {
-    setPlaySoundOnNewMessage(value)
     updateUserSettings({ playSoundOnNewMessage: value })
   }
 ```

What do you think of this approach?

Reply:
great :)